RecoParticleFlow/PFClusterProducer/plugins/Basic2DGenericPFlowPositionCalc.h

```diff
@@ -8,16 +8,58 @@
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 #include "RecoParticleFlow/PFClusterProducer/interface/CaloRecHitResolutionProvider.h"
+#include <tuple>
 
 class Basic2DGenericPFlowPositionCalc : public PFCPositionCalculatorBase {
  public:
   Basic2DGenericPFlowPositionCalc(const edm::ParameterSet& conf) :
     PFCPositionCalculatorBase(conf),    
     _posCalcNCrystals(conf.getParameter<int>("posCalcNCrystals")),
-    _logWeightDenom(1./conf.getParameter<double>("logWeightDenominator")),
     _minAllowedNorm(conf.getParameter<double>("minAllowedNormalization"))
-
   {  
+
+    std::vector<int> detectorEnum;
+    std::vector<int> depths;
+    std::vector<double> logWeightDenom;
+    std::vector<float> logWeightDenomInv;
+
+    if(conf.exists("logWeightDenominatorByDetector")) {
```

Comment:
conf.exists are strongly discouraged and should be replaced by a valid fillDescriptions method.
It may be easier to define a required config parameter that would tell which configuration path should be picked.
Perhaps it's even better to enforce that logWeightDenominatorByDetector is the way to go without the compatibility support of the old configuration.

Reply:
I prefer to keep the conf.exists as this is already done originally multiple times in the same file.
The else condition is not only needed to support the old configuration but also to not  modify the config of ECAL/HF/PS/HO.
Since the mantainance of the RecoParticleFlow/PFClusterProducer it's not an HCAL responsability so i will not go in the fillDescriprition business.